src/dsp_tools/commands/validate_data/models/api_responses.py

```diff
@@ -26,10 +26,16 @@ def hlist(self) -> str:
         return f'"hlist=<{self.list_iri}>"'
 
     def shacl_nodes(self) -> str:
-        sorted_nodes = sorted(self.nodes)
-        output = [f'"{x}"' for x in sorted_nodes]
-        node_str = " ".join(output)
+        escaped_nodes = [self._escape(x) for x in sorted(self.nodes)]
+        node_str = " ".join(escaped_nodes)
         return f"( {node_str} )"
 
     def shacl_list(self) -> str:
-        return f'( "{self.list_name}" )'
+        return f"( {self._escape(self.list_name)} )"
+
+    def escaped_list_name(self) -> str:
+        return self.list_name.replace("\\", "\\\\").replace("'", "\\'").replace('"', '\\"')
+
+    def _escape(self, in_str: str) -> str:
+        replace_problems = in_str.replace("\\", "\\\\").replace("'", "\\'").replace('"', '\\"')
+        return '"' + replace_problems + '"'
```

Comment:
**Issue**: Escaping should reusable and context aware

I propose we have dedicated functions for escaping rather than adding it as methods to a data class.

Having it on the data class is convenient to use but hard to reuse

Also, escaping in general is heavily depending on the context of where it is applied, i.e. it is probably different for Turtle as compared in a json context.

Reply:
I agree, my intention was to put it at the only place where it is needed in this milestone, and look at the correct serialisation in the last milestone and analyse it in detail. But if you prefer to take a look now I am fine with that too.